platforms/gke/base/core/workloads/lws/main.tf

```diff
@@ -0,0 +1,98 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+locals {
+  kubeconfig_directory = "${path.module}/../../../kubernetes/kubeconfig"
+  kubeconfig_file      = "${local.kubeconfig_directory}/${local.kubeconfig_file_name}"
+
+  manifests_directory         = "${local.manifests_directory_root}/namespace/lws-system"
+  namespace_directory         = "${local.manifests_directory_root}/namespace"
+  version_manifests_directory = "${path.module}/manifests/lws-${var.lws_version}"
+}
+
+data "local_file" "kubeconfig" {
+  filename = local.kubeconfig_file
+}
+
+resource "terraform_data" "namespace" {
+  input = {
+    manifests_dir = local.namespace_directory
+  }
+
+  provisioner "local-exec" {
+    command     = <<EOT
+mkdir -p ${self.input.manifests_dir} && \
+cp -r templates/namespace-lws-system.yaml ${self.input.manifests_dir}/
+EOT
+    interpreter = ["bash", "-c"]
+    working_dir = path.module
+  }
+
+  triggers_replace = {
+    manifests_dir = local.namespace_directory
+  }
+}
+
+module "kubectl_apply_namespace" {
+  depends_on = [
+    terraform_data.namespace,
+  ]
+
+  source = "../../../modules/kubectl_apply"
+
+  kubeconfig_file             = data.local_file.kubeconfig.filename
+  manifest                    = "${local.namespace_directory}/namespace-lws-system.yaml"
+  manifest_includes_namespace = true
+}
+
+resource "terraform_data" "manifests" {
+  input = {
+    manifests_dir         = local.manifests_directory
+    version_manifests_dir = local.version_manifests_directory
+    version               = var.lws_version
+  }
+
+  provisioner "local-exec" {
+    command     = <<EOT
+mkdir -p ${self.input.version_manifests_dir} && \
+mkdir -p ${self.input.manifests_dir} && \
+wget https://github.com/kubernetes-sigs/lws/releases/download/v${self.input.version}/manifests.yaml -O ${self.input.version_manifests_dir}/manifests.yaml && \
+cp -r templates/workload/* ${self.input.version_manifests_dir}/ && \
+cp -r ${self.input.version_manifests_dir}/* ${self.input.manifests_dir}/
+EOT
+    interpreter = ["bash", "-c"]
+    working_dir = path.module
+  }
+
+  triggers_replace = {
```

Comment:
This will not trigger if you update a manifest. Consider adding the hash of the contents of manifests as an additional trigger.

Example:

https://github.com/GoogleCloudPlatform/accelerated-platforms/blob/0f61da5de37f698e56d82fef2cf1e11f95216f99/platforms/gke/base/use-cases/federated-learning/terraform/config_management/main.tf#L92-L97

Reply:
The intent was only to trigger when the version was changed.  I don't intend for these Terraservices to be used for upgrades or modification. Ideally they are just there to deploy the workload and generated initial manifest that can then be put into version control and/or ConfigSync/Argo.

Reply:
/shrug